substrate/frame/transaction-payment/asset-conversion-tx-payment/src/lib.rs

```diff
@@ -285,7 +285,8 @@ impl<T: Config> TransactionExtension<T::RuntimeCall> for ChargeAssetTxPayment<T>
 where
 	T::RuntimeCall: Dispatchable<Info = DispatchInfo, PostInfo = PostDispatchInfo>,
 	BalanceOf<T>: Send + Sync + From<u64>,
-	T::AssetId: Send + Sync,
+	<BalanceOf<T> as codec::HasCompact>::Type: DecodeWithMemTracking,
+	T::AssetId: DecodeWithMemTracking + Send + Sync,
```

Comment:
Then we can remove this.

Reply:
Answered above.

Reply:
Done, but please notice that doing `BalanceOf: HasDecodeWithMemTrackingCompact` lead to more changes